### ISSUE TITLE
[SCT_EXTRACT_METRIC] Add weighted-Maximum a posteriori feature

### ISSUE THREAD

**Issue by tanguyduval** [neuropoly/spinalcordtoolbox#1018]
If some voxels are less reliable than other we should be able to submit an image of weights. For instance axon diameter or g-Ratio are totally unstable in the CSF. Moreover if the spinal cord touch the bones the metric value at the border can be totally wrong...

To do that change the problem as follows:
`````
``
 
y = AX ==> Wy = WAX
```````
With
```````
W(i,i) = wi (weigth of voxel i).
W(i,j) = 0 (for i /= j)
```````

Basically in all equations replace:
``````
A^tA --> A^tWA
y ---> Wy
``````

Thanks!!

**Comment by slevyrosetti** [comment]
I find this feature interesting but I am worried that it won't be used enough to be implemented: indeed, it requires the creation of a mask with particular weights for each voxel. To me this seems really specific to one case and is unlikely to be used... What do you think @jcohenadad ?

**Comment by jcohenadad** [comment]
yes, add flag: -mask-weighted

**Comment by slevyrosetti** [comment]
This feature was added in branch `sl_extract_metric`, commit 0bce426.

**Comment by jcohenadad** [comment]
fixed in PR #1040
